neuralprophet/forecaster.py

```diff
@@ -3242,7 +3244,24 @@ def _reshape_raw_predictions_to_forecst_df(self, df, predicted, components, pred
                     forecast_rest = components[comp][1:, self.n_forecasts - 1, j]
                     yhat = np.concatenate(([np.NaN] * self.max_lags, forecast_0, forecast_rest))
                     if prediction_frequency is not None:
-                        forecast_rest = components[comp][1:, :, j]
+                        for key, value in prediction_frequency.items():
+                            if key == "daily":
+                                freq = 24
+                            elif key == "weekly":
+                                freq = 7
+                            elif key == "yearly":
+                                freq = 12
+                            elif key == "monthly":
+                                freq = 30
+                            elif key == "hourly":
+                                freq = 60
+                            else:
+                                raise ValueError(f"Unknown prediction frequency {key}")
+                            # Check if the next forecast for the component overlaps with the previous one
+                            if self.n_forecasts > freq:
+                                forecast_rest = components[comp][1:, -(self.n_forecasts - freq) :, j]
+                            else:
+                                forecast_rest = components[comp][1:, :, j]
```

Comment:
For Trend, Seasonality and so on: Flatten the forecasts for those components according to the prediction frequency